app/controllers/funding/payment_schedules_controller.rb

```diff
@@ -2,13 +2,15 @@
 
 module Funding
   class PaymentSchedulesController < BaseFundingController
+    before_action :redirect, only: [:show]
```

Comment:
maybe something like `before_action :redirect_to_schedule_path, unless: -> { academic_year.present? }, only: [:show]` ? then your method could just have the redirect